receiver/elasticsearchreceiver/scraper.go

```diff
@@ -95,12 +97,33 @@ func (r *elasticsearchScraper) scrape(ctx context.Context) (pmetric.Metrics, err
 
 	now := pcommon.NewTimestampFromTime(time.Now())
 
+	err := r.getVersion(ctx)
+	if err != nil {
+		return pmetric.NewMetrics(), fmt.Errorf("unable to determine version of elasticsearch: %w", err)
```

Comment:
I'm not convinced it's a good idea to fail based on this. It's only needed for one metric right? Is it possible that this would fail but other metrics could be retrieved?

Reply:
The base request to get the version would have a curl request as `http://localhost:9200`. The other client request build on top of the original request as  `http://localhost:9200/_nodes/_all/stats/{nodeStatsMetrics}/{nodeStatsIndexMetrics}` or `http://localhost:9200/_cluster/health`. I can't imagine a scenario where the version request fails and the other 2 node/cluster requests returns a payload. 

What I can do instead though if you are still not convinced is revert the last change and instead add a nil check `if r.version != nil && r.version.GreaterThanOrEqual(es10) {` that way if the version is ever nil then the second portion won't trigger and panic.

Reply:
Updated. Added a nil check so that way less assumptions are made on rather request combinations can be valid/invalid. Now only one metric is not recorded if there is a problem with getting the version.

Reply:
I like this better, even though it probably is unlikely to make a difference. It is more robust this way.